### PR TITLE
CHORE: MERGE VERSION-15 AND STAGING INTO ATTENDANCE BRANCH

### PR DESCRIPTION
**Pull request by NoBoneZ**
## Is this a Feature, Chore or Bug?
- [] Feature
- [X] Chore
- [] Bug

## Clearly and concisely describe the feature, chore or bug.
Merge Version-15 and staging branch into attendance

## Analysis and design (optional)
Analyse and attach the design documentation

## Solution description
Describe your code changes in detail for reviewers.

## Is there a business logic within a doctype?
    - [] Yes
    - [] No

## Output screenshots (optional)
Post the output screenshots, if a UI is affected or added due to this feature.

## Areas affected and ensured
List out the areas affected by your code changes.

## Is there any existing behavior change of other features due to this code change?
Mention Yes or No. If Yes, provide the appropriate explanation.

## Did you test with the following dataset?
- [] Existing Data
- [] New Data

## Was child table created?
    - [] is attachment required?
        did you test attachment
## Did you delete custom field?
    - [] Yes
    - [] No
        If yes, did you write a delete patch?

## Is patch required?
- [] Yes
- [] No
    ## Was the patch test?

## Which browser(s) did you use for testing?
  - [] Chrome
  - [] Safari
  - [] Firefox

### CODE REVIEW

File: one_fm/public/js/doctype_js/attendance_request.js [hunk 1]
```diff
@@ -1,8 +1,17 @@
 frappe.ui.form.on('Attendance Request', {
   refresh: (frm)=>{
     frm.trigger('check_workflow');
+<<<<<<< HEAD
```
Comment (pjamsheer) [inline one_fm/public/js/doctype_js/attendance_request.js]: Please check this line number 4, 7 ,.........

File: one_fm/public/js/doctype_js/employee_checkin.js [hunk 2]
```diff
@@ -20,9 +20,19 @@ frappe.ui.form.on('Employee Checkin', {
 });
 
 
+<<<<<<< HEAD
```
Comment (pjamsheer) [inline one_fm/public/js/doctype_js/employee_checkin.js]: Please review this file

File: one_fm/one_fm/custom/shift_request.json [hunk 3]
```diff
@@ -1,4 +1,5 @@
 {
+<<<<<<< HEAD
```
Comment (pjamsheer) [inline one_fm/one_fm/custom/shift_request.json]: Please check this file

File: one_fm/public/js/doctype_js/timesheet.js [hunk 4]
```diff
@@ -1,13 +1,11 @@
...
     employee: function(frm) {
         set_approver(frm)
-    },
-    attendance_by_timesheet: function(frm) {
-        set_approver(frm)
     }
 })
 
 function set_approver(frm){
+<<<<<<< HEAD
```
Comment (pjamsheer) [inline one_fm/public/js/doctype_js/timesheet.js]: Please check this file

File: one_fm/public/swagger/one_fm_v1.json
Comment (pjamsheer) [inline one_fm/public/swagger/one_fm_v1.json:1]: Please check this file

**Review by pjamsheer — CHANGES_REQUESTED** [review]

File: one_fm/public/js/doctype_js/timesheet.js (hunk 4 above)
Comment (NoBoneZ) [reply one_fm/public/js/doctype_js/timesheet.js]: @pjamsheer , changes have been made

File: one_fm/one_fm/custom/shift_request.json (hunk 3 above)
Comment (NoBoneZ) [reply one_fm/one_fm/custom/shift_request.json]: @pjamsheer , changes have been made

File: one_fm/public/js/doctype_js/attendance_request.js (hunk 1 above)
Comment (NoBoneZ) [reply one_fm/public/js/doctype_js/attendance_request.js]: @pjamsheer , changes have been made

File: one_fm/public/js/doctype_js/employee_checkin.js (hunk 2 above)
Comment (NoBoneZ) [reply one_fm/public/js/doctype_js/employee_checkin.js]: @pjamsheer , changes have been made

File: one_fm/public/js/doctype_js/attendance_request.js [hunk 5]
```diff
@@ -1,8 +1,12 @@
 frappe.ui.form.on('Attendance Request', {
   refresh: (frm)=>{
     frm.trigger('check_workflow');
+    set_update_request_btn(frm);
```
Comment (pjamsheer) [inline one_fm/public/js/doctype_js/attendance_request.js]: This method is not defined(we have removed it)

**Review by pjamsheer — CHANGES_REQUESTED** [review]

File: one_fm/public/js/doctype_js/attendance_request.js (hunk 5 above)
Comment (NoBoneZ) [reply one_fm/public/js/doctype_js/attendance_request.js]: @pjamsheer , i have resolved this

File: one_fm/overrides/shift_request.py [hunk 6]
```diff
@@ -567,31 +663,19 @@ def validate_approver(self):
...
+    approver = get_approver(doc.employee)
+    approver_user_id = get_employee_user_id(approver)
+    other_approvers = []
+    if doc.department == "Operations - ONEFM":
+        other_approvers =[user.approver for user in frappe.get_all("Department Approver", filters={"parent": doc.department, "parentfield": "shift_request_approver"}, fields=["approver"]) if user.approver != employee_user_id and user.approver != approver_user_id]
+
+    if approver_user_id:
+        return [approver_user_id] + other_approvers
+
+<<<<<<< HEAD
```
Comment (pjamsheer) [inline one_fm/overrides/shift_request.py]: Please check this file

**Review by pjamsheer — CHANGES_REQUESTED** [review]

File: one_fm/overrides/shift_request.py (hunk 6 above)
Comment (NoBoneZ) [reply one_fm/overrides/shift_request.py]: @pjamsheer resolved

**Review by pjamsheer — APPROVED** [review]

**Pull request merged**